openfl/utilities/verifdataset/local_data_source.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+""" This module contains the LocalDataSource class. """
+from hashlib import sha384
+from pathlib import Path
+
+from openfl.utilities.verifdataset.data_source import DataSource, DataSourceType
+
+
+class LocalDataSource(DataSource):
+    """This class represents a local data source."""
+
+    def __init__(self, source_path: Path, hash_func=sha384, maxDatasetSize=0):
+        super().__init__(DataSourceType.LOCAL)
+        self.source_path = Path(source_path)
+        self.hash_func = hash_func
+        self.max_dataset_size = maxDatasetSize
+
+    def enumerate_objects(self, base_dir: Path):
+        """Walk through the data directory and collect all files."""
```

Comment:
do you get the real path ? os.path.realpath()

Reply:
yes, it is done when register dataset before creating dataset_info.
The datasource itself does not hold the full path but only relative path, which is calculated before creating the datasources (against the real path of course), so it is not really testable but anyway I added some tests as we use in the test the same logic of getting the base path and relative paths.